deps/llvm.mk

```
@@ -70,7 +74,7 @@ LLVM_EXPERIMENTAL_TARGETS :=
 LLVM_CFLAGS :=
 LLVM_CXXFLAGS :=
 LLVM_CPPFLAGS :=
-LLVM_LDFLAGS :=
+LLVM_LDFLAGS := "$(build_shlibdir)/libz.$(SHLIB_EXT)" # Hacky way to force zlib to be found when linking against libLLVM and sysroot is set
```

@giordano perhaps you know whether this is reasonable? or alternatively know someone else to ask about this?

---

@vchuravy @gbaraldi maybe?

---

Might be cleaner just to force it to pass the shlibdir? (until the llvm bug is fixed)
```suggestion
LLVM_LDFLAGS := "-L$(build_shlibdir)" # hacky way to force zlib to be found when linking against libLLVM and sysroot is set
```

---

I would prefer fixing this in the LLVM build system and carrying a patch, but Jameson's proposal also looks okay

---

I was just rebuilding llvm 20 and it looked correct already. Any idea if this is already fixed upstream though?

---

I'm just going off of the logs for a recent build (https://buildkite.com/julialang/yggdrasil/builds/20102#0196b19f-bf86-4672-a692-ce66874bbb1a) which show that it only using the absolute path to libz with the sysroot set. It appears to be fine with both that and libzstd?